internal/services/channel_account_service.go

```diff
@@ -0,0 +1,186 @@
+		return fmt.Errorf("getting the number of channel account already stored: %w", err)
+	}
+
+	numOfChannelAccountsToCreate := number - currentChannelAccountNumber
+	// The number of channel accounts stored is sufficient.
+	if numOfChannelAccountsToCreate <= 0 {
+		return nil
+	}
+
+	distributionAccountPublicKey, err := s.DistributionAccountSignatureClient.GetAccountPublicKey(ctx)
+	if err != nil {
+		return fmt.Errorf("getting distribution account public key: %w", err)
+	}
+
+	ops := make([]txnbuild.Operation, 0, numOfChannelAccountsToCreate)
+	channelAccountsToInsert := []*channelaccounts.ChannelAccount{}
+	for range numOfChannelAccountsToCreate {
+		kp, err := keypair.Random()
+		if err != nil {
+			return fmt.Errorf("generating random keypair for channel account: %w", err)
+		}
+
+		encryptedPrivateKey, err := s.PrivateKeyEncrypter.Encrypt(ctx, kp.Seed(), s.EncryptionPassphrase)
+		if err != nil {
+			return fmt.Errorf("encrypting channel account private key: %w", err)
+		}
+
+		ops = append(ops, &txnbuild.CreateAccount{
+			Destination:   kp.Address(),
+			Amount:        "1",
+			SourceAccount: distributionAccountPublicKey,
+		})
+		channelAccountsToInsert = append(channelAccountsToInsert, &channelaccounts.ChannelAccount{
+			PublicKey:           kp.Address(),
+			EncryptedPrivateKey: encryptedPrivateKey,
+		})
+	}
+
+	if err = s.submitCreateChannelAccountsOnChainTransaction(ctx, distributionAccountPublicKey, ops); err != nil {
+		return fmt.Errorf("inserting channel accounts: %w", err)
+	}
+
+	if err = s.ChannelAccountStore.BatchInsert(ctx, s.DB, channelAccountsToInsert); err != nil {
+		return fmt.Errorf("inserting channel accounts: %w", err)
+	}
+
+	return nil
+}
+
+func (s *channelAccountService) submitCreateChannelAccountsOnChainTransaction(ctx context.Context, distributionAccountPublicKey string, ops []txnbuild.Operation) error {
+	distributionAccount, err := s.HorizonClient.AccountDetail(horizonclient.AccountRequest{AccountID: distributionAccountPublicKey})
+	if err != nil {
+		return fmt.Errorf("getting account detail of channel account: %w", err)
+	}
+
+	tx, err := txnbuild.NewTransaction(
+		txnbuild.TransactionParams{
+			SourceAccount:        &distributionAccount,
+			IncrementSequenceNum: true,
+			Operations:           ops,
+			BaseFee:              s.BaseFee,
+			Preconditions:        txnbuild.Preconditions{TimeBounds: txnbuild.NewTimeout(300)},
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("building transaction: %w", err)
+	}
+
+	signedTx, err := s.DistributionAccountSignatureClient.SignStellarTransaction(ctx, tx, distributionAccountPublicKey)
+	if err != nil {
+		return fmt.Errorf("signing transaction: %w", err)
+	}
+
+	hash, err := signedTx.Hash(s.DistributionAccountSignatureClient.NetworkPassphrase())
+	if err != nil {
+		return fmt.Errorf("getting transaction hash: %w", err)
+	}
+
+	_, err = s.HorizonClient.SubmitTransaction(signedTx)
+	if err != nil {
+		if hError := horizonclient.GetError(err); hError != nil {
```

Comment:
Maybe we could start thinking of creating a submission service for HTTP requests with retry logic in it (not in this PR for sure). Wdyt of that idea?